TLM/TLM/Manager/Impl/ExtSegmentManager.cs

```diff
@@ -154,6 +194,26 @@ public class ExtSegmentManager
             return true;
         }
 
+        private uint[] CalculateLanes(ref NetSegment segment) {
+            if (!segment.IsValid()) {
+                return null;
+            }
+
+            var info = segment.Info;
+            int laneCount = info.m_lanes.Length;
```

Comment:
Some stuff can be unexpectedly `null`, particularly with older roads from workshop.

```csharp
byte laneCount = (byte)(segmentInfo?.m_lanes?.Length ?? 0);
```

By storing it as byte (I think there's max 32 or 64 lanes limit per segment - @krzychu124 will know) you can use `byte` in the `for` loop later on.

Reply:
I'm not sure if byte is a good idea, compiler will use cast operation everywhere possible. See mentioned example _(yes that's exactly the result of compilation for the one liner (as `Release` target)_):

![image](https://user-images.githubusercontent.com/19638970/162579164-04c1798f-d6bf-4507-b5da-82594ab94067.png)

later on it even casts `byte` to `int` to get NetInfo.Lane from array...
![image](https://user-images.githubusercontent.com/19638970/162579313-88b8fddb-1899-4891-845b-02a06dcc6e9d.png)

---
Anyways, CO expects `null` for `m_lanes` sometimes, `null` for `Info` is directly because of an issue with the asset (incomplete initialization because of bug or missing dependencies)

Reply:
If `IsValid` isn't checking for null, maybe it should. But I'll add a check on `m_lanes`.

Reply:
I checks only for validity from simulation perspective (segment was created and not marked as deleted)

Reply:
> `null` for `Info` is directly because of an issue with the asset

Are we saying, then, that this is not a normative scenario that needs to be checked for?

Reply:
I think we (TM:PE) are checking for that case everywhere when working with segments outside of normal methods. I'm pretty sure we don't check for null if NetSegment comes as param beacuse in 99% it would be redundant

Reply:
If we're going to be processing every segment, we should be checking here and then make this the hub of everything we want to know about lanes so we don't need to check in all the other code.

Reply:
null can happen only if there is a bug. in any case in the rest of CS and TMPE code we have checked for null so it might be wise to do this here too.
its not performance critical part of the code.